bee-api/bee-rest-api/src/types/responses.rs

```diff
@@ -0,0 +1,247 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::types::{
+    body::BodyInner,
+    dtos::{LedgerInclusionStateDto, MessageDto, OutputDto, PeerDto, ReceiptDto},
+};
+
+use serde::{Deserialize, Serialize};
+
+/// Response of GET /api/v1/info.
+/// Returns general information about the node.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct InfoResponse {
+    pub name: String,
```

Comment:
Seeing as these fields are public, is it worth documenting them? Same applies to public enum variants (eg. `Error`).

Reply:
@rufsam You need to add `#![deny(missing_docs, warnings)]` to the lib.rs as all other crates anyway

Reply:
I've thought about it a bit, and came to the conclusion that we probably shouldn't enforce `#![deny(warnings)]` for our released crates. Let will make for example "deprecated" warnings by a next version of the compiler hard errors immediatedly. Basically on each new compiler release we would fear people cannot compile Bee until we fixed all of our crates.

Reply:
@thibault-martinez it is added?

Reply:
I've thought about it a bit, and came to the conclusion that we probably shouldn't enforce `#![deny(warnings)]` for our released crates. That will make for example "deprecated" warnings by a next version of the compiler hard errors immediatedly. Basically on each new compiler release we would fear people cannot compile Bee until we fixed all of our crates.

Reply:
I started adding `#![deny(warnings)]` because it was already in the other crates but I agree with you, I don't think we should.

Reply:
alright, I'll remove it in `bee-network` as well

Reply:
Also, we deny warnings in the CI, so WE would never be able to merge warnings anyway, I think that's sufficient.